db/migrate/20160322195653_move_switch_host_to_jointable.rb

```diff
@@ -0,0 +1,25 @@
+class MoveSwitchHostToJointable < ActiveRecord::Migration[5.0]
+  class Host < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    has_many :switches,
+             :class_name => 'MoveSwitchHostToJointable::Switch'
+  end
+
+  class Switch < ActiveRecord::Base
+    belongs_to :host,
+               :class_name => 'MoveSwitchHostToJointable::Host'
+  end
+
+  class HostSwitch < ActiveRecord::Base
+  end
+
+  def up
+    say_with_time('Populating host_switches table with (host.id, switch.id)') do
+      Host.all.each do |host|
```

Comment:
```ruby
Host.includes(:switches).find_each do |host|
```

Preload the switches association (to avoid N+1 queries)...and use [find_each](http://api.rubyonrails.org/classes/ActiveRecord/Batches.html) to find in batches.  This is probably not necessary here but it's useful to use that convention in case we have thousands of rows so we can avoid unneeded queries and not instantiate all the things in memory at once.